From patchwork Tue Jan 20 18:09:03 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alexey Charkov <alchark@gmail.com>
X-Patchwork-Id: 2186851
X-Patchwork-Delegate: narmstrong@baylibre.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@legolas.ozlabs.org
Authentication-Results: legolas.ozlabs.org;
	dkim=pass (2048-bit key;
 unprotected) header.d=gmail.com header.i=@gmail.com header.a=rsa-sha256
 header.s=20230601 header.b=JMAINTNr;
	dkim-atps=neutral
Authentication-Results: legolas.ozlabs.org;
 spf=pass (sender SPF authorized) smtp.mailfrom=lists.denx.de
 (client-ip=2a01:238:438b:c500:173d:9f52:ddab:ee01; helo=phobos.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de; receiver=patchwork.ozlabs.org)
Received: from phobos.denx.de (phobos.denx.de
 [IPv6:2a01:238:438b:c500:173d:9f52:ddab:ee01])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange x25519)
	(No client certificate requested)
	by legolas.ozlabs.org (Postfix) with ESMTPS id 4dwb435KMzz1xrM
	for <incoming@patchwork.ozlabs.org>; Wed, 21 Jan 2026 05:11:15 +1100 (AEDT)
Received: from h2850616.stratoserver.net (localhost [IPv6:::1])
	by phobos.denx.de (Postfix) with ESMTP id 6120483A6A;
	Tue, 20 Jan 2026 19:11:05 +0100 (CET)
Authentication-Results: phobos.denx.de;
 dmarc=pass (p=none dis=none) header.from=gmail.com
Authentication-Results: phobos.denx.de;
 spf=pass smtp.mailfrom=u-boot-bounces@lists.denx.de
Authentication-Results: phobos.denx.de;
	dkim=pass (2048-bit key;
 unprotected) header.d=gmail.com header.i=@gmail.com header.b="JMAINTNr";
	dkim-atps=neutral
Received: by phobos.denx.de (Postfix, from userid 109)
 id EA02C8313B; Tue, 20 Jan 2026 19:09:39 +0100 (CET)
X-Spam-Checker-Version: SpamAssassin 3.4.2 (2018-09-13) on phobos.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-1.1 required=5.0 tests=BAYES_00,DKIM_SIGNED,
 DKIM_VALID,DKIM_VALID_AU,DKIM_VALID_EF,FORGED_GMAIL_RCVD,FREEMAIL_FROM,
 RCVD_IN_DNSWL_BLOCKED,SPF_HELO_NONE,SPF_PASS autolearn=no
 autolearn_force=no version=3.4.2
Received: from mail-wr1-x431.google.com (mail-wr1-x431.google.com
 [IPv6:2a00:1450:4864:20::431])
 (using TLSv1.3 with cipher TLS_AES_128_GCM_SHA256 (128/128 bits))
 (No client certificate requested)
 by phobos.denx.de (Postfix) with ESMTPS id 0D3348063E
 for <u-boot@lists.denx.de>; Tue, 20 Jan 2026 19:09:38 +0100 (CET)
Authentication-Results: phobos.denx.de;
 dmarc=pass (p=none dis=none) header.from=gmail.com
Authentication-Results: phobos.denx.de;
 spf=pass smtp.mailfrom=alchark@gmail.com
Received: by mail-wr1-x431.google.com with SMTP id
 ffacd0b85a97d-42fbc305552so4121060f8f.0
 for <u-boot@lists.denx.de>; Tue, 20 Jan 2026 10:09:38 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=gmail.com; s=20230601; t=1768932577; x=1769537377; darn=lists.denx.de;
 h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
 :mime-version:subject:date:from:from:to:cc:subject:date:message-id
 :reply-to; bh=B8Yyt0SfLUmQ8CiuNr+p+99DftmooKje1R2T+ThDa3M=;
 b=JMAINTNrX5/rR+kdQRZNxbDID4xT0ymjgeBOsWo8bG2YgV+gobn+3JtjOO/iTFbu/T
 uNFc8UVFXrm/E0ggMFdVV0TNZu+3MOGE3E+2OacyiKKnXZXzd2YNJOZGDlEvYXHDARv7
 NSTscBB2nQLpuKCURinHv8eAlXQFtGFdoPlx5lYAcD1/hoS4nRdrVprGImR9a5CCV/AF
 0uAR4/uvj6Oetdo5rrOCE17NTjegOtIrJsY+bxFt8ThCtR9HAS2mdhWmTo36AiSbncFT
 Lzku6E4Ld65lIfZ5NSaO84PSx4SM7uzEl1WubNMi0SVFgKawd39W58o1GDhAYoVSWLqT
 n7sw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20230601; t=1768932577; x=1769537377;
 h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
 :mime-version:subject:date:from:x-gm-gg:x-gm-message-state:from:to
 :cc:subject:date:message-id:reply-to;
 bh=B8Yyt0SfLUmQ8CiuNr+p+99DftmooKje1R2T+ThDa3M=;
 b=qeA5rzMkjk2LUTDVJ1uXICdmlzj1dImdEZXH1BMtrkDwBw+YY6P2InDMKWQkMGteQj
 Aq9VK6pjB9biMKAry3C+/Bw26Olyf2ndA1Z4C/Ib8IGoiv3v2IIZj7E8M4B4drV04ZsC
 snSotdXpe6f6U/fQo60OAnn4fwDFIgH5suRgw9mlhSXrT9C0EYOEi5+q+/4Ab62KlZxn
 iiu7E9S0hz3bcteRl/kmygF/3bO/NNLkFsODcYM/CzqeB6f4nuevMyOKha+IygkUOtfL
 BWF1RvFFFa5E3LuBZqZs60cq1IPnMOvGp1+hrdPJHvDf6spgKsbjd4TaDi/XUFZhONoo
 AaKg==
X-Gm-Message-State: AOJu0YxxMHdlULNufHhYrw1gw14Y0Asaw7wohD9yAmbZRHSrP0T0vmlE
 AxsSbOyjwJ+MbgVClmMiT3xxCWGzcklv22xh8AaGCyvd0CS8elih1NIW
X-Gm-Gg: AZuq6aKEzKeMGmcREKxCClI4+zfYtBzVSt2tWp9P/PYDv7VCX5S4w5qa7hgrsaACqhG
 KXQ+Qzm0yKfqDTyElVyXfmaeXVJgzQuJp8WKVJmWHYc7zbNg1jLLrIoA+4vYY87KNlDFPlUqspH
 ZnN2iJLgnEoeRCasyDya4uJ8PzXbYdFJvGDOB003N59AgsiqI7SmOURJgLY6oWqcz6+Xs+yWzpQ
 rdCAOKLupea6b5UQ2sQkuSxPWtBMTwLc1wrQ5NfMiLoyfPeUMt8A8odJy8EmDvBJwLLoJtnEP+0
 U9QzjjpuM3ToGCNwfYLAuxHZttF054DfmsNYk2skHHPjHsnz8ODg7PlH8fFtVAqV2LsCugh+zvc
 qIHzGBwE1quEBkZzJIYWtNu2lBX3R4zNWvmUXAi1OPqnWdJ6dW+6Q5O0FS94brum23nAn4Xxaf8
 WAXoafxdtcS04lgmvow9gf+ZkkcDkzptgRIHlX3Xeqvnckc+OiJ+7GNqxaH98tIu0=
X-Received: by 2002:a05:6000:2f83:b0:432:5b81:480 with SMTP id
 ffacd0b85a97d-4356a03d2demr21753895f8f.24.1768932577111;
 Tue, 20 Jan 2026 10:09:37 -0800 (PST)
Received: from alchark-surface.localdomain (bba-83-110-134-52.alshamil.net.ae.
 [83.110.134.52]) by smtp.gmail.com with ESMTPSA id
 ffacd0b85a97d-43569998240sm30168043f8f.43.2026.01.20.10.09.29
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Tue, 20 Jan 2026 10:09:36 -0800 (PST)
From: Alexey Charkov <alchark@gmail.com>
Date: Tue, 20 Jan 2026 22:09:03 +0400
Subject: [PATCH v5 2/4] reset: rockchip: make device resets available in
 SPL
MIME-Version: 1.0
Message-Id: <20260120-rk3576-ufs-v5-2-0edb61b301b7@gmail.com>
References: <20260120-rk3576-ufs-v5-0-0edb61b301b7@gmail.com>
In-Reply-To: <20260120-rk3576-ufs-v5-0-0edb61b301b7@gmail.com>
To: u-boot@lists.denx.de, Shawn Lin <shawn.lin@rock-chips.com>
Cc: Tom Rini <trini@konsulko.com>, Kever Yang <kever.yang@rock-chips.com>,
 Jonas Karlman <jonas@kwiboo.se>, Heiko Stuebner <heiko@sntech.de>,
 Neil Armstrong <neil.armstrong@linaro.org>,
 Bhupesh Sharma <bhupesh.linux@gmail.com>,
 Neha Malcom Francis <n-francis@ti.com>, Greg Malysa <malysagreg@gmail.com>,
 Vasileios Bimpikas <vasileios.bimpikas@analog.com>,
 Arturs Artamonovs <arturs.artamonovs@analog.com>,
 Utsav Agarwal <utsav.agarwal@analog.com>,
 Ian Roberts <ian.roberts@timesys.com>,
 Nathan Barrett-Morrison <nathan.morrison@timesys.com>,
 Peng Fan <peng.fan@nxp.com>,
 "Kory Maincent (TI.com)" <kory.maincent@bootlin.com>,
 Alif Zakuan Yuslaimi <alif.zakuan.yuslaimi@altera.com>,
 Simon Glass <sjg@chromium.org>, Yao Zi <me@ziyao.cc>,
 Stefan Roese <stefan.roese@mailbox.org>,
 Mattijs Korpershoek <mkorpershoek@kernel.org>,
 Sumit Garg <sumit.garg@kernel.org>, Heiko Schocher <hs@nabladev.com>,
 Anshul Dalal <anshuld@ti.com>, Andre Przywara <andre.przywara@arm.com>,
 Heinrich Schuchardt <xypron.glpk@gmx.de>,
 Quentin Schulz <quentin.schulz@cherry.de>, Andrew Davis <afd@ti.com>,
 Hrushikesh Salunke <h-salunke@ti.com>,
 Dario Binacchi <dario.binacchi@amarulasolutions.com>,
 Dinesh Maniyam <dinesh.maniyam@altera.com>,
 Ilias Apalodimas <ilias.apalodimas@linaro.org>,
 Philippe Reynes <philippe.reynes@softathome.com>,
 Marek Vasut <marek.vasut@mailbox.org>,
 Philipp Tomsich <philipp.tomsich@vrull.eu>,
 Marek Vasut <marek.vasut+renesas@mailbox.org>,
 Igor Belwon <igor.belwon@mentallysanemainliners.org>,
 Tuyen Dang <tuyen.dang.xa@renesas.com>,
 Xuhui Lin <xuhui.lin@rock-chips.com>, Johan Jonker <jbx6244@gmail.com>,
 Lukasz Czechowski <lukasz.czechowski@thaumatec.com>,
 Dhruva Gole <d-gole@ti.com>, Patrice Chotard <patrice.chotard@foss.st.com>,
 Paul Barker <paul.barker.ct@bp.renesas.com>,
 Gabriel Fernandez <gabriel.fernandez@foss.st.com>,
 Leo Yu-Chi Liang <ycliang@andestech.com>,
 Christian Marangi <ansuelsmth@gmail.com>,
 Elaine Zhang <zhangqing@rock-chips.com>,
 Joseph Chen <chenjh@rock-chips.com>, Alexey Charkov <alchark@gmail.com>
X-Mailer: b4 0.14.3
X-Developer-Signature: v=1; a=openpgp-sha256; l=1756; i=alchark@gmail.com;
 h=from:subject:message-id; bh=qz9kqohcFj8GIIYB3FowNNQ2uu/rY2sUMGjsyhFXlTk=;
 b=owGbwMvMwCW2adGNfoHIK0sZT6slMWTmHzm1sUPrUsHepCNnLnlPvbx78wfBwMvf/GMnvXPu3
 3LtfcDq3I6JLAxiXAyWYoosc78tsZ1qxDdrl4fHV5g5rEwgQ6RFGhiAgIWBLzcxr9RIx0jPVNtQ
 z9BQx1jHiIGLUwCmeqMVwz99oc+3tX8WBDw6qVr/fJnwoWd/2WPPswU/mPOu3+XVnM9LGBkaNPd
 f41py9Ne6o0s0BUvM/YI9bir5XDNX2yG35tPS5UocAA==
X-Developer-Key: i=alchark@gmail.com; a=openpgp;
 fpr=9DF6A43D95320E9ABA4848F5B2A2D88F1059D4A5
X-Mailman-Approved-At: Tue, 20 Jan 2026 19:11:03 +0100
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.39
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <https://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>
X-Virus-Scanned: clamav-milter 0.103.8 at phobos.denx.de
X-Virus-Status: Clean

Enable the Rockchip reset controller driver in SPL to allow resetting
attached devices like UFS during early boot.

Signed-off-by: Alexey Charkov <alchark@gmail.com>
---
 drivers/reset/Kconfig  | 9 +++++++++
 drivers/reset/Makefile | 2 +-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/reset/Kconfig b/drivers/reset/Kconfig
index 74c267dfc4e5..d35f3904e404 100644
--- a/drivers/reset/Kconfig
+++ b/drivers/reset/Kconfig
@@ -99,6 +99,15 @@ config RESET_ROCKCHIP
 	  though is that some reset signals, like I2C or MISC reset multiple
 	  devices.
 
+config SPL_RESET_ROCKCHIP
+	bool "SPL reset controller driver for Rockchip SoCs"
+	depends on SPL_DM_RESET && ARCH_ROCKCHIP && CLK
+	default y
+	help
+	  Support for the reset controller on Rockchip SoCs in SPL. Select this
+	  if you observe any reset-related warnings or errors when booting SPL,
+	  such as when using UFS storage
+
 config RESET_HSDK
 	bool "Synopsys HSDK Reset Driver"
 	depends on DM_RESET && TARGET_HSDK
diff --git a/drivers/reset/Makefile b/drivers/reset/Makefile
index ee5b009d1341..dc7d3b610e6e 100644
--- a/drivers/reset/Makefile
+++ b/drivers/reset/Makefile
@@ -16,7 +16,7 @@ obj-$(CONFIG_RESET_BCM6345) += reset-bcm6345.o
 obj-$(CONFIG_RESET_UNIPHIER) += reset-uniphier.o
 obj-$(CONFIG_RESET_AST2500) += reset-ast2500.o
 obj-$(CONFIG_RESET_AST2600) += reset-ast2600.o
-obj-$(CONFIG_RESET_ROCKCHIP) += reset-rockchip.o rst-rk3528.o rst-rk3576.o rst-rk3588.o
+obj-$(CONFIG_$(PHASE_)RESET_ROCKCHIP) += reset-rockchip.o rst-rk3528.o rst-rk3576.o rst-rk3588.o
 obj-$(CONFIG_RESET_MESON) += reset-meson.o
 obj-$(CONFIG_RESET_SOCFPGA) += reset-socfpga.o
 obj-$(CONFIG_RESET_MEDIATEK) += reset-mediatek.o
